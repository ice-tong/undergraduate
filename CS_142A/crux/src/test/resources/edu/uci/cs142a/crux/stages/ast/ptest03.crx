// test nested array indexing

array x : int[4];
array y : int[4][5];

func main() : void {
  let x[y[1][2]] = x[y[1][1]];
}
