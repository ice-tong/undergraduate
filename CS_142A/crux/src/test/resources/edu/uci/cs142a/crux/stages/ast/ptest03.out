ast.DeclarationList(3,0)
  ast.ArrayDeclaration(3,0)[Symbol(x)]
  ast.ArrayDeclaration(4,0)[Symbol(y)]
  ast.FunctionDefinition(6,0)[Symbol(main), []]
    ast.StatementList(7,2)
      ast.Assignment(7,2)
        ast.Index(7,8)
          ast.AddressOf(7,6)[x]
          ast.Dereference(7,8)
            ast.Index(7,13)
              ast.Index(7,10)
                ast.AddressOf(7,8)[y]
                ast.LiteralInt(7,10)[1]
              ast.LiteralInt(7,13)[2]
        ast.Dereference(7,19)
          ast.Index(7,21)
            ast.AddressOf(7,19)[x]
            ast.Dereference(7,21)
              ast.Index(7,26)
                ast.Index(7,23)
                  ast.AddressOf(7,21)[y]
                  ast.LiteralInt(7,23)[1]
                ast.LiteralInt(7,26)[1]
